modules/iasRtdProvider.js

```diff
@@ -0,0 +1,122 @@
+import { submodule } from '../src/hook.js';
+import { getGlobal } from '../src/prebidGlobal.js';
+import * as utils from '../src/utils.js';
+import { ajaxBuilder } from '../src/ajax.js';
+
+/** @type {string} */
+const MODULE_NAME = 'realTimeData';
+const SUBMODULE_NAME = 'ias';
+
+/**
+ * Module init
+ * @param {Object} provider
+ * @param {Object} userConsent
+ * @return {boolean}
+ */
+export function init(config, userConsent) {
+  return true;
+}
+
+function stringifySlotSizes(sizes) {
+  let result = '';
+  if (utils.isArray(sizes)) {
+    result = sizes.reduce((acc, size) => {
+      acc.push(size.join('.'));
+      return acc;
+    }, []);
+    result = '[' + result.join(',') + ']';
+  }
+  return result;
+}
+
+function stringifySlot(bidRequest) {
+  const id = bidRequest.code;
+  const ss = stringifySlotSizes(bidRequest.sizes);
+  const p = bidRequest.bids[0].params.adUnitPath;
+  const slot = { id, ss, p };
+  const keyValues = utils.getKeys(slot).map(function (key) {
+    return [key, slot[key]].join(':');
+  });
+  return '{' + keyValues.join(',') + '}';
+}
+
+function stringifyWindowSize() {
+  return [window.innerWidth || -1, window.innerHeight || -1].join('.');
+}
+
+function stringifyScreenSize() {
+  return [(window.screen && window.screen.width) || -1, (window.screen && window.screen.height) || -1].join('.');
+}
+
+function getPageLevelKeywords(response) {
+  let result = {};
+  shallowMerge(result, response.brandSafety);
+  result.fr = response.fr;
+  result.custom = response.custom;
+  return result;
+}
+
+function shallowMerge(dest, src) {
+  utils.getKeys(src).reduce((dest, srcKey) => {
```

Comment:
Can you add a check against src to ensure it is defined? If the wrong id is configured utils.getKeys(src) errors out since brandSafety does not exist

Reply:
@mmoschovas I added the validation